src/components/FormValidation/FormValidation.js

```diff
@@ -0,0 +1,46 @@
+import React, { useCallback } from "react";
+
+// hook for form control
+export function useForm() {
+    const [values, setValues] = React.useState({});
+
+    const handleChange = (event) => {
+        const target = event.target;
+        const value = target.value;
+        const name = target.name;
+        setValues({ ...values, [name]: value });
+    };
+
+    return { values, handleChange, setValues };
+}
+
+// hook for form control and form validation
+export function useFormWithValidation() {
```

Comment:
any custom hook should be in a separate file in folder `hooks` to show other developers that you have them. It's easy to find them visually